lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -244,103 +229,139 @@ abstract class BaseAdapter {
   void _addWheelEventListener(html.EventListener handler) {
     final dynamic eventOptions = js_util.newObject();
     final html.EventListener jsHandler = js.allowInterop((html.Event event) => handler(event));
-    _nativeListeners['wheel'] = jsHandler;
+    _BaseAdapter._nativeListeners['wheel'] = jsHandler;
     js_util.setProperty(eventOptions, 'passive', false);
     js_util.callMethod(
-      domRenderer.glassPaneElement,
+      glassPaneElement,
       'addEventListener', <dynamic>[
         'wheel',
         jsHandler,
         eventOptions
       ]
     );
-
   }
 }
 
-const int _kPrimaryMouseButton = 0x1;
-const int _kSecondaryMouseButton = 0x2;
-
-int _pointerButtonFromHtmlEvent(html.Event event) {
-  if (event is html.PointerEvent) {
-    final html.PointerEvent pointerEvent = event;
-    return pointerEvent.button == 2
-        ? _kSecondaryMouseButton
-        : _kPrimaryMouseButton;
-  } else if (event is html.MouseEvent) {
-    final html.MouseEvent mouseEvent = event;
-    return mouseEvent.button == 2
-        ? _kSecondaryMouseButton
-        : _kPrimaryMouseButton;
+typedef void _PointerStateDataCallback({
+  @required ui.PointerChange change,
+  @required int buttons,
+});
+
+class _PointerStateManager {
+  int _pressedButtons = 0;
+
+  // Transform html.PointerEvent.buttons to Flutter's PointerEvent buttons.
+  int _htmlButtonsToFlutterButtons(int buttons) {
+    // Flutter's button definition conveniently matches that of JavaScript
+    // from primary button (0x1) to forward button (0x10), which allows us to
+    // avoid transforming it bit by bit.
+    return buttons & _kButtonsMask;
+  }
+
+  void handleDown({@required _PointerStateDataCallback dataCallback, @required int buttons}) {
```

Comment:
I wish Dart had support for tuples. We would have avoided the `dataCallback` and just returned a `Tuple<ui.PointerChange, int>`.

Reply:
We can always create a class for that, but I think it's not as flexible as callbacks, since some handlers are stateful, and returning the list would be a little more complicated.

Reply:
On a second thought, I'll make it a list to address the issue mentioned in https://github.com/flutter/engine/pull/14378#discussion_r357272311